radar-auth/src/main/java/org/radarcns/auth/authentication/AlgorithmLoader.java

```diff
@@ -0,0 +1,81 @@
+package org.radarcns.auth.authentication;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import org.radarcns.auth.exception.TokenValidationException;
+import org.radarcns.auth.security.jwk.JavaWebKey;
+import org.radarcns.auth.security.jwk.JavaWebKeySet;
+import org.radarcns.auth.token.validation.ECTokenValidationAlgorithm;
+import org.radarcns.auth.token.validation.RSATokenValidationAlgorithm;
+import org.radarcns.auth.token.validation.TokenValidationAlgorithm;
+import org.radarcns.auth.token.validation.deprecated.DeprecatedEcTokenValidationAlgorithm;
+
+public class AlgorithmLoader {
+
+    private static final List<TokenValidationAlgorithm> algorithmList = Arrays.asList(
+            new ECTokenValidationAlgorithm(),
+            new RSATokenValidationAlgorithm());
+    private static final List<TokenValidationAlgorithm> supportedAlgorithmsForPublicKeys =
+            Arrays.asList(
+            new DeprecatedEcTokenValidationAlgorithm(),
```

Comment:
Both the deprecated and new EcToken validation should be supported for public keys.

Reply:
Sorry. I am not sure whether the variable names were misleading. RSA and new EC are supported for JWKs and RSA and DeprecatedEC are supported for loading public-keys configured in files, which i consider the "deprecated way". Did you mean that new EC and Deprecated EC should be supported for loading public-keys from config-file?
Having both EC versions in the public-key support may lead to some confusions, since we do not know how to decide whether a public-key is for new EC or Deprecated EC. We can only decide whether it is EC key or not. We can indeed try till there is a successful loading. But I think we also don't have a need for that.  
I made the implementation based on the concept of, if there is a EC public key provided in config file, it is generated from Deprecated EC method and all the new tokens/implementations should depend on public key endpoint to fetch public-keys. This way we can be sure that after a single iteration of refreshing old tokens all of the tokens have migrated to the new EC method and the public-key endpoint can be used for validation.